File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/TrainingRun.js

// Define collection and schema for TrainingRun
export interface TrainingRun {
    runLabel:
	type : string
    startedAt:
	type : Date
    completedAt:
	type : Date
    Experiment:
	type : Schema.Types.ObjectId
    ModelVersion:
	type : Schema.Types.ObjectId
    InputDatasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    Features:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Feature' }]
    RunMetrics:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RunMetric' }]
    RunParameters:
 	type : [{ type: Schema.Types.ObjectId, ref: 'RunParameter' }]
    Status:
 	type : String
#
    collection: 'trainingRuns'
}

File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/TrainingRun.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// TrainingRun Declaration
//==============================================================
type TrainingRun struct {
    gorm.Model
     RunLabel                                    string
    StartedAt                                                            time.Time
    CompletedAt                                                            time.Time
    ExperimentId         *uint
    Experiment           *Experiment `gorm:"foreignKey:ExperimentId"`
    ModelVersionId         *uint
    ModelVersion           *ModelVersion `gorm:"foreignKey:ModelVersionId"`
     InputDatasets           []DataSet `gorm:"foreignKey:InputDatasetsFromTrainingRunId"`
     Features           []Feature `gorm:"foreignKey:FeaturesFromTrainingRunId"`
     RunMetrics           []RunMetric `gorm:"foreignKey:RunMetricsFromTrainingRunId"`
     RunParameters           []RunParameter `gorm:"foreignKey:RunParametersFromTrainingRunId"`
    Status                      TrainingStatus

// parent associations as their child

}

